ConfigSpace/util.py

```diff
@@ -708,3 +708,68 @@ def get_cartesian_product(value_sets: list[tuple], hp_names: list[str]) -> list[
         unchecked_grid_pts.popleft()
 
     return checked_grid_pts
+
+
+def remove_hyperparameter(name: str, configuration_space: ConfigurationSpace) -> ConfigurationSpace:
+    """
+    Returns a new configuration space with the hyperparameter removed.
+
+    Parameters
+    ----------
+    name: str
+        Name of the hyperparameter to remove
+
+    configuration_space: :class:`~ConfigSpace.configuration_space.ConfigurationSpace`
+        Configuration space from which to remove the hyperparameter.
+
+    Returns
+    -------
+    :class:`~ConfigSpace.configuration_space.Configuration`
+        A new configuration space without the hyperparameter
+    """
+    if name not in configuration_space._hyperparameters:
+        raise ValueError(f"{name} not in {configuration_space}")
+
+    hp_to_remove = configuration_space.get_hyperparameter(name)
+    hps = [
+        copy(hp)  # type: ignore
+        for hp in configuration_space.get_hyperparameters()
+        if hp.name != name
+    ]
+
+    conditions = [
+        cond
+        for cond in configuration_space.get_conditions()
+        if hp_to_remove not in cond.get_referenced_hyperparameters()
+    ]
+    forbiddens = [
+        forbidden
+        for forbidden in configuration_space.get_forbiddens()
+        if hp_to_remove not in forbidden.get_referenced_hyperparameters()
+    ]
+
+    if isinstance(configuration_space.random, np.random.RandomState):
+        new_seed = configuration_space.random.randint(2**32 - 1)

```

Comment:
We should probably copy the `RandomState`, rather then create a new one: https://stackoverflow.com/a/63875402/5332072

Reply:
Very interesting.  I have always thought it is not possible to copy RandomState.